src/utils/wrap_thread.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef WRAP_THREAD_HPP
+#define WRAP_THREAD_HPP
+
+#include <chrono>
+#include <future>
+#include <thread>
+
+/// More advanced then std::thread. Will join itself on destruction to avoid certain UBs. Has debug and stats.
+/// @owner hb
+class wrap_thread {
+	public:
+		wrap_thread() noexcept;
+
+		// std::chrono::duration destroy_time = 0
+		template<typename Function, typename ...Args>
```

Comment:
///< this creates an thread that must be joined by caller by using .join() . If caller doesn't do that then destructor will abort/terminate the program.

and change the BEHAVIOUR of destructor to do that.